src/components/reader-page-image-list.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import React, { Component, Fragment } from 'react';
+import type { Page } from 'poketo';
+import mostCommon from 'array-most-common';
+import type { PageDimensions } from '../types';
+import ReaderPageImage from '../components/reader-page-image';
+
+type PageListProps = {
+  pages: Page[],
+};
+
+type PageListState = {
+  defaultPageWidth?: number,
+  defaultPageHeight?: number,
+};
+
+function serializeDimension(dimension: PageDimensions) {
+  return `${dimension.width}:${dimension.height}`;
+}
+
+function unserializeDimension(dimension: string) {
+  const [width, height] = dimension.split(':');
+  return { width, height };
+}
+
+function getMostCommonPageDimensions(
+  dimensions: PageDimensions[],
+): PageDimensions {
+  if (dimensions.length < 2) {
+    // If there's less than 3 pages, getting the most common dimensions is
+    // likely to be wrong, since the first few scans of a chapter often have
+    // title/introductory pages which are different ratios.
+    return { width: undefined, height: undefined };
+  }
+
+  const serializedDimensions = dimensions.map(serializeDimension);
+  const mostCommonSerialized = mostCommon(serializedDimensions);
+  const mostCommonDimension = unserializeDimension(mostCommonSerialized);
+
+  return mostCommonDimension;
+}
+
+export default class ReaderPageImageList extends Component<
+  PageListProps,
+  PageListState,
+> {
+  state = {
+    defaultPageWidth: undefined,
+    defaultPageHeight: undefined,
+  };
+
+  pageSizes: PageDimensions[] = [];
```

Comment:
The `pageSizes` variable isn't included in the component's state because adding to it would cause re-renders — we only want re-renders when `defaultPageWidth` or `defaultPageHeight` changes though.

By keeping our list separate, we only re-render when our defaults have changed.